cmake/sanitize.cmake

```diff
@@ -10,9 +10,7 @@ set (SAN_FLAGS "${SAN_FLAGS} -g -fno-omit-frame-pointer -DSANITIZER")
 
 if (SANITIZE)
     if (SANITIZE STREQUAL "address")
-        # LLVM-15 has a bug in Address Sanitizer, preventing the usage of 'sanitize-address-use-after-scope',
```

Comment:
This breaks asan builds for users still building with Clang 15. Afaik, there was a policy of supporting the last two stable clang versions. But I wouldn't mind too much, we could also force >= clang 16, see `CLANG_MINIMUM_VERSION` in cmake/tools.cmake.